types/utils/either.ts

```diff
@@ -0,0 +1,3 @@
+type Only<T, U> = { [P in keyof T]: T[P] } & { [P in keyof U]?: never };
```

Comment:
This breaks down if T and U share a key, causing the overall type to be `never`:

<img width="342" alt="Screenshot 2023-11-28 at 10 46 15 AM" src="https://github.com/Azure/azure-functions-nodejs-library/assets/11282622/69347bc7-5d0f-4251-8da2-0c3a60d375c1">

Reply:
I found a modification of `Only` that accounts for these cases:

```
type Only<T, U> = {   [P in keyof T]: T[P] } & Omit<{   [P in keyof U]?: never }, keyof T>
```